### PR TITLE
ER-630 QA review re-submitted vacancies

### PR DESCRIPTION
**Pull request by chrisjensenuk**
- Created `VacancyComparerService` which returns a list of changed fields between two vacancies.  May be of use elsewhere in future.

Have deleted all the private beta referral logic:
- Table on the QA dashboard
- Refer link on the Review
- Removed text boxes from the Refer view
- Removed supporting functions and properties

### CODE REVIEW

File: src/QA/QA.Web/Orchestrators/ReviewOrchestrator.cs [hunk 1]
```diff
@@ -103,52 +60,39 @@ public async Task<ReviewViewModel> GetReviewViewModelAsync(Guid reviewId, Vacanc
...
             await EnsureUserIsAssignedAsync(review, user.UserId);
 
-            var vm = await _mapper.MapFromVacancy(review.VacancySnapshot);
+            var vm = await _mapper.Map(review);
 
             return vm;
         }
 
-        public async Task<ReviewViewModel> GetReviewViewModelAsync(ReviewEditModel model, VacancyUser user)
+        public async Task<ReviewViewModel> GetReadonlyReviewViewModelAsync(Guid reviewId, VacancyUser user)
```
Comment (leewadhams) [inline src/QA/QA.Web/Orchestrators/ReviewOrchestrator.cs]: User argument isn't used in method.

File: src/Shared/Recruit.Vacancies.Client/Application/CommandHandlers/CreateVacancyReviewCommandHandler.cs [hunk 2]
```diff
@@ -74,10 +82,33 @@ private VacancyReview BuildNewReview(Vacancy vacancy, int previousReviewCount, D
...
+
+            var comparison = _vacancyComparerService.Compare(vacancy, previousReview.VacancySnapshot);
+
+            return comparison.Fields
+                .Where(f => f.AreEqual == false)
+                .Select(f => f.FieldName)
+                .ToList();
+        }
+
+        private VacancyReview GetPreviousReferredVacancyReviewAsync(IEnumerable<VacancyReview> allReviewsForVacancy)
```
Comment (leewadhams) [inline src/Shared/Recruit.Vacancies.Client/Application/CommandHandlers/CreateVacancyReviewCommandHandler.cs]: Does this need to be names "Async"?

File: src/Shared/Recruit.Vacancies.Client/Application/Services/VacancyComparer/VacancyComparaerResult.cs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+﻿using System.Collections.Generic;
+
+namespace Esfa.Recruit.Vacancies.Client.Application.Services.VacancyComparer
+{
+    public class VacancyComparerResult
```
[the remark this answers is not in the document]
Comment (chrisjensenuk) [reply src/Shared/Recruit.Vacancies.Client/Application/Services/VacancyComparer/VacancyComparaerResult.cs]: 👍

File: src/QA/QA.Web/Orchestrators/ReviewOrchestrator.cs (hunk 1 above)
Comment (chrisjensenuk) [reply src/QA/QA.Web/Orchestrators/ReviewOrchestrator.cs]: 👍

File: src/Shared/Recruit.Vacancies.Client/Application/CommandHandlers/CreateVacancyReviewCommandHandler.cs (hunk 2 above)
Comment (chrisjensenuk) [reply src/Shared/Recruit.Vacancies.Client/Application/CommandHandlers/CreateVacancyReviewCommandHandler.cs]: 👍

**Comment by chrisjensenuk** [comment]
@leewadhams have pushed fixes for you comments.

**Pull request merged**